plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/DeployAssemblyEntryRemoveJob.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.newproject;
+
+import com.google.cloud.tools.eclipse.util.status.StatusUtil;
+import com.google.common.annotations.VisibleForTesting;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.jst.j2ee.refactor.listeners.J2EEElementChangedListener;
+import org.eclipse.wst.common.componentcore.ComponentCore;
+import org.eclipse.wst.common.componentcore.resources.IVirtualComponent;
+import org.eclipse.wst.common.componentcore.resources.IVirtualFolder;
+
+/**
+ * Removes entries from WTP's deployment assembly whose source path matches a given path.
+ */
+// Intended to be used by our wizards during project creation. See
+// https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1997#issuecomment-310707483
+class DeployAssemblyEntryRemoveJob extends Job {
+
+  private final IProject project;
+  private final IPath sourcePath;
+
+  // Cannot use "join()", as "NonSystemJobSuspender" makes it return immediately without running it.
```

Comment:
Oh interesting.  What a tangled web we weave…  I wonder if we shouldn't make NonSystemJobSuspender specific to ConvertJob.

But this comment seems to belong elsewhere (it's actually for the users of this job).  And instead of this boolean, callers can use `getState() == Job.NONE` to see if it's been executed.

An alternative is to make `DeployAssemblyEntryRemoveJob` to be a system job — at least for the `CreateAppEngineWtpProject` case.